### DISCUSSION TITLE
Translate/transform bug for long distances

### DISCUSSION THREAD

**Discussion by javi-mr** [pyvista/pyvista#4128] [Q&A]
Hello!

When I try to translate a triangulated surface (PolyData) to a far position ([438384.09, 4478309.04, 661.21]) the final mesh is completely deformed. Do you know why?

Is there any method to translate the surface by setting the new center of the mesh?

Thank you!

**Comment by banesullivan** [comment]
Oh wow. This is indeed odd

```py
import pyvista as pv

mesh = pv.Sphere()
trans = mesh.translate([438384.09, 4478309.04, 661.21])
trans.plot()
```
![screenshot (5)](https://user-images.githubusercontent.com/22067021/225378692-43a93546-36a5-4e05-95b0-20ad56db0c8d.png)


And we get the same thing with:

```py
import pyvista as pv

mesh = pv.Sphere()
mesh.points += [438384.09, 4478309.04, 661.21]
mesh.plot()
```

This is unexpected...

**Reply by banesullivan** [reply]
We start to see deformation around 1e5:

```
import pyvista as pv

mesh = pv.Sphere()
mesh.points += [1e5, 1000.0, 1000.0]
mesh.plot()
```
![screenshot (7)](https://user-images.githubusercontent.com/22067021/225381424-892b6c1b-16db-4583-8112-9ae10e625e3b.png)

**Reply by banesullivan** [reply]
This looks like a VTK issue as a similar thing is happening in ParaView:

![Screen Shot 2023-03-15 at 10 49 44 AM](https://user-images.githubusercontent.com/22067021/225381871-c92efe1f-2bb6-4398-981b-80677663bcd4.png)

**Reply by banesullivan** [reply]
This issue in this example at least seems that the points are single/float32 and thus have only 8 digits to work with. Once we get to a value like `4478309.04`, the variability of the mesh's points in 8,9,10 digit places and thus we loose that precision.

Changing to float64 mitigates, but you will then run into the same problem after 16 digits

```py
import pyvista as pv

mesh = pv.Sphere()
mesh.points_to_double()
trans = mesh.translate([438384.09, 4478309.04, 661.21])
trans.plot()
```
![screenshot (8)](https://user-images.githubusercontent.com/22067021/225388159-b7e69162-6b2f-42d5-8005-12b2e1162b35.png)

**Reply by javi-mr** [reply]
16 digits are enough for me! Let me know if you solve this odd behavior regardless of the number of digits. 

Additionally I would like to know if it's possible to move the surface by defining the center (without translating the surface). I'm trying to use a slider (pl.add_slider_widget) to move an object along its trajectory in the plotter and it's more useful to define the different positions than to translate it.

Thank you Bane :)

**Reply by luizenger** [reply]
Instead of redefining the center point of the existing object, couldn't you just delete the previous object and create a new one at the specified location?
The translate( ) method does this anyway if I understood the code correctly (the newly generated mesh substitutes the existing one if inplace=True)

**Reply by banesullivan** [reply]
You may be able to apply a transform to the actor without modifying the points of the mesh: https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Actor.user_matrix.html#user-matrix

**Reply by JasperBoogers** [reply]
Is it possible to convert all data arrays to double precision, so to extend the result ``points_to_double()`` to all arrays? for example, all cell normal vectors and areas? I found very little documentation on this topic online, including the existence of ``points_to_double()``

**Reply by banesullivan** [reply]
This is not currently implemented but I agree it should be. Would you be open to contributing this feature?

**Reply by JasperBoogers** [reply]
I would be honoured, but not for another month at least. I first have to finish a master's degree (which I could not have done without Pyvista 😄 )
